core-scala/src/main/scala/com/baeldung/scala/futures/Futures.scala

```diff
@@ -0,0 +1,90 @@
+package com.baeldung.scala.futures
+
+import java.util.concurrent.{Executor, ExecutorService, ForkJoinPool}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+object Futures extends App {
+  val executor: Executor = new ForkJoinPool(4)
+  val executorService: ExecutorService = new ForkJoinPool(4)
+
+  ExecutionContext.fromExecutor(executor)
+  ExecutionContext.fromExecutorService(executorService)
+
+  def getToken(): String = "secretToken"
+
+  val tokenF: Future[String] = Future(getToken())
+  val token: String = Await.result(tokenF, Duration.Inf)
+
+  val numberF: Future[Int] = Future.successful(5)
+  val oddNumberF: Future[Int] = numberF.filter(_ % 2 != 0) // complete with 5
```

Comment:
These comments are not helping the code to make sense. I think they would be better ahead of the code they describe.